homeassistant/components/rainmachine/switch.py

```diff
@@ -390,6 +394,14 @@ async def async_turn_on_when_active(self, **kwargs: Any) -> None:
             )
         )
 
+        # When the zone starts, calculate when it is expected to be done and set the
```

Comment:
How does this work if the zone is started from the app or via a program?

Reply:
Hmm, good point. Let me think on it.